prime-router/src/testIntegration/resources/datatests/mappinginventory/xon/xon-to-organization-xon10-populated.hl7

```diff
@@ -0,0 +1 @@
+MSH|^~\&#|||||20230501102531-0400||ORU^R01^ORU_R01|3003786103_4988249_33033||2.5.1|||||USA|UNICODE UTF-8||||Org^A^123^Check Digit^C1^Assigning Authority&2.1.4.1&ISO^MD^Hospital A&2.16.840.1.113883.9.11&ISO^^Hospital B
```

Comment:
This sure is bare bones, which is good, but when I try to use the CLI to test the conversions outside of test cases, I get all kinds of validation issues. In various iterations of trying to run this MSH through the CLI I've gotten:
```
19:10:36.012 [main][1] WARN  gov.cdc.prime.router.fhirengine.translation.hl7.FhirToHl7Converter - Could not set HL7 value for spec MSH-22-3 for element id-number
ca.uhn.hl7v2.model.DataTypeException: ca.uhn.hl7v2.validation.ValidationException: Validation failed: Primitive value '123' requires to be empty because it is withdrawn from the current HL7 version and should not be used
```
and
```
19:36:02.084 [main][1] WARN  gov.cdc.prime.router.fhirengine.translation.hl7.FhirToHl7Converter - Could not set HL7 value for spec MSH-22-4 for element identifier-check-digit
ca.uhn.hl7v2.model.DataTypeException: ca.uhn.hl7v2.validation.ValidationException: Validation failed: Primitive value 'Check Digit' requires to be empty or a number with optional decimal digits
```
It seems like the `FHIRToHl7Converter`is going to prevent us from populating XON.3. I haven't spent enough time looking into it to know if that's just a CLI thing or if the same validation code is used in the running application. But yea we should probably figure out what validation is used in the CLI vs the application vs the translation tests, and maybe try to get those synced up. Even if the mappings work in the tests it might be wasted effort if the application would never be able to utilize those mappings due to this kind of validation.

To recreate what I'm seeing, you can copy/paste a valid hl7 message into `junk/input.hl7`, swap out MSH with the contents of this file, tweak things as desired, and run the following:
```
./prime fhirdata --input-file junk/input.hl7 --output-file junk/output.fhir --output-format FHIR
./prime fhirdata --input-file junk/output.fhir -s metadata/hl7_mapping/ORU_R01/ORU_R01-base.yml --output-file junk/output.hl7 --output-format HL7
```

Reply:
Hmm... let me play around with it, but I didn't think those messages actually caused any issues with the output?  Though agree that having them get produced is messy.

Reply:
Ah so, I actually don't think that the the errors you're highlighting is what prevents the CLI from finishing correctly.
```
Exception in thread "main" gov.cdc.prime.router.fhirengine.translation.hl7.RequiredElementException: Required element patient-information conditional was false or value was empty.
        at gov.cdc.prime.router.fhirengine.translation.hl7.FhirToHl7Converter.processElement$prime_router(FhirToHl7Converter.kt:148)
        at gov.cdc.prime.router.fhirengine.translation.hl7.FhirToHl7Converter.processSchema(FhirToHl7Converter.kt:123)
        at gov.cdc.prime.router.fhirengine.translation.hl7.FhirToHl7Converter.processSchema$default(FhirToHl7Converter.kt:104)
        at gov.cdc.prime.router.fhirengine.translation.hl7.FhirToHl7Converter.convert(FhirToHl7Converter.kt:94)
        at gov.cdc.prime.router.cli.ProcessFhirCommands.convertToHl7(ProcessFhirCommands.kt:145)
        at gov.cdc.prime.router.cli.ProcessFhirCommands.run(ProcessFhirCommands.kt:99)
        at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:198)
        at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:211)
        at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:18)
        at com.github.ajalt.clikt.core.CliktCommand.parse(CliktCommand.kt:400)
        at com.github.ajalt.clikt.core.CliktCommand.parse$default(CliktCommand.kt:397)
        at com.github.ajalt.clikt.core.CliktCommand.main(CliktCommand.kt:415)
        at com.github.ajalt.clikt.core.CliktCommand.main(CliktCommand.kt:440)
        at gov.cdc.prime.router.cli.MainKt.main(main.kt:297)
```

Is also the exception I see.

The test codes actually uses `ORU_R01-test.yml` rather than `ORU_R01-base.yml` and makes some required elements not required.

Reply:
Ah that's a good call. To clarify, I was just putting this MSH onto another valid message to get around those required elements, but the difference between `ORU_R01-test` and `ORU_R01-base` explains a lot. Try swapping `251` with `27` in `ORU_R01-test` and rerunning the CLI conversions, and that should show what I'm talking about, and it seems like as a result of it, MSH.22.3 or MSH.22.4 get set. In the case of MSH.22.4, I think we can resolve it by using a valid input for that field, but for MSH.22.3, it doesn't seem like v2.7 will ever let us set that.

Reply:
Confirmed... just need to redo how we initially create the HL7 when converting from FHIR and these exceptions go away.